File: problems/268-missing-number/python/268_missing_number.py
from typing import List


class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        curr_sum: int = sum(nums)
        expected: int = int(len(nums) * (len(nums) + 1) / 2)
        return expected - curr_sum


def missingNumberTest(nums: List[int], solution: int) -> None:
    result: int = Solution().missingNumber(nums)
    if result == solution:
        print(f" + 'Test ML({nums}) == {solution}' passed!")
    else:
        print(f" - 'Test ML({nums}) == {solution}' failed with value {result}!")


if __name__ == "__main__":
    missingNumberTest([3, 0, 1], 2)
    missingNumberTest([0, 1], 2)
    missingNumberTest([9, 6, 4, 2, 3, 5, 7, 0, 1], 8)

File: problems/268-missing-number/csharp/268_missing_number.cs
using System.Collections.Generic;
using System.Linq;
using System;

public static class Program
{
  public static void MissingNumberTest(int[] nums, int solution)
  {
    int result = new Solution().MissingNumber(nums);
    if (result == solution) { Console.WriteLine(" + Test 'MN([{0}]) == {1}' passed!", string.Join(",", nums), solution); }
    else { Console.WriteLine(" - Test 'MN([{0}]) == {1}' failed with value {2}!", string.Join(",", nums), solution, result); }
  }
  public static void Main(string[] args)
  {
    MissingNumberTest(new int[] { 3, 0, 1 }, 2);
    MissingNumberTest(new int[] { 0, 1 }, 2);
    MissingNumberTest(new int[] { 9, 6, 4, 2, 3, 5, 7, 0, 1 }, 8);
  }
}

public class Solution
{
  public int MissingNumber(int[] nums)
  {
    int sum = nums.ToList().Sum();
    int expected = nums.Length * (nums.Length + 1) / 2;
    return expected - sum;
  }
}